File: src/quest_editor/actions/EditLongDescriptionAction.ts
import { QuestEditAction } from "./QuestEditAction";

export class EditLongDescriptionAction extends QuestEditAction<string> {
    readonly description = "Edit long description";

    undo(): void {
        this.quest.set_long_description(this.old);
    }

    redo(): void {
        this.quest.set_long_description(this.new);
    }
}

File: web/src/main/kotlin/world/phantasmal/web/questEditor/actions/EditLongDescriptionAction.kt
package world.phantasmal.web.questEditor.actions

import world.phantasmal.web.core.actions.Action
import world.phantasmal.web.questEditor.models.QuestModel

class EditLongDescriptionAction(
    private val quest: QuestModel,
    private val newLongDescription: String,
    private val oldLongDescription: String,
) : Action {
    override val description: String = "Edit long description"

    override fun execute() {
        quest.setLongDescription(newLongDescription)
    }

    override fun undo() {
        quest.setLongDescription(oldLongDescription)
    }
}
